test/images/agnhost/Dockerfile

```diff
@@ -12,6 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# builder image
+# populate the fake registry with images
+FROM golang:1.24 AS preparer
+
+RUN go install github.com/google/go-containerregistry/cmd/crane@latest && \
+    apt-get update && apt-get install -y jq
+
+COPY fakeregistryserver/prepare_image.sh /prepare_image.sh
+RUN chmod +x /prepare_image.sh
+
+# for now it's only pause:3.10.1, but we can download more images if needed
+RUN /prepare_image.sh pause 3.10.1 /registry
```

Comment:
Here's a thought: Let's always _download_ the current pause image tag, we can track this with build/dependencies.yaml to make sure it stays up to date (should add that config in this PR), but let's expose a *fixed* tag on the local private registry process so we don't have confusion coordinating which tag the e2e tests should pull.

Otherwise it will get confusing because the pause tag pulled by the test will need to match the version in the current agnhost image after that image has been built and published + adopted, which will intermittently skew ... 

We could call it `$host/pause:test` or something, and just have the script always locally stage the manifest at `test`.